dbms/src/Storages/AlterCommands.cpp

```diff
@@ -165,6 +166,32 @@ std::optional<AlterCommand> AlterCommand::parse(const ASTAlterCommand * command_
 
         return command;
     }
+    else if (command_ast->type == ASTAlterCommand::ADD_CONSTRAINT)
+    {
+        AlterCommand command;
+        command.constraint_decl = command_ast->constraint_decl;
+        command.type = AlterCommand::ADD_CONSTRAINT;
+
+        const auto & ast_constraint_decl = command_ast->constraint_decl->as<ASTConstraintDeclaration &>();
+
+        command.constraint_name = ast_constraint_decl.name;
+
+        command.if_not_exists = command_ast->if_not_exists;
+
+        return command;
+    }
+    else if (command_ast->type == ASTAlterCommand::DROP_CONSTRAINT)
+    {
+        if (command_ast->clear_column)
```

Comment:
Tried to avoid copying and merged three similar sections. Check it https://github.com/yandex/ClickHouse/pull/5273/files#diff-6c02eeac4c6ef1353865d6162925543a